src/util/taproot.rs

```diff
@@ -780,23 +784,38 @@ impl LeafVersion {
     // or an opcode that is not valid as the first opcode).
     // The values that comply to this rule are the 32 even values between
     // 0xc0 and 0xfe and also 0x66, 0x7e, 0x80, 0x84, 0x96, 0x98, 0xba, 0xbc, 0xbe
-    pub fn from_u8(ver: u8) -> Result<Self, TaprootError> {
-        if ver & TAPROOT_LEAF_MASK == ver && ver != 0x50 {
-            Ok(LeafVersion(ver))
-        } else {
-            Err(TaprootError::InvalidTaprootLeafVersion(ver))
+    pub fn from_u8(version: u8) -> Result<Self, TaprootError> {
```

Comment:
I think this should be called `from_consensus` or something like that to make it clear it doesn't mean 0 => Tapscript.

Reply:
`from_consensus_u8`?;)

Reply:
Not sure which is better. As long as it's obvious what is meant by it I'm fine.

Reply:
@apoelstra I think that `from_consensus_u*` makes sense only when there are multiple representation of the same thing,  like sig hash with `u8` and `u32`

Reply:
Makes sense. I'm happy with whatever name y'all want.